src/troute_model.py

```diff
@@ -27,6 +28,15 @@ def __init__(self, bmi_cfg_file):
                      '_segment_attributes', '_waterbody_attributes', '_network',
                      '_data_assimilation', '_fvd', '_lakeout']
         
+        self.task_times = {                                     # Creating a dictionary to record time for each section #
+            'network_time': 0,
+            'forcing_time' : 0,
+            'output_time' : 0,
+            'run_time' :0,
+            'data_assimilation_time' : 0,
+
+        }
+        self.main_start_time = time.time()
```

Comment:
main_start_time is called here but never referenced again.

Because t-route is being called via BMI functions and other formulations will also be running, we'll need to rethink how we log the total time. Calling time.time() here and then again after all is said and done would actually record the timing of t-route PLUS all of the other formulations. I think if we cleverly place all the other timing calls we can simply add them all up when we finalize() the BMI instance of t-route.

Reply:
Self.main_start_time can be used to log total time if needed. But I don't think its necessary here since we breaking it down. I'll delete it.